src/ansi-c/ansi_c_entry_point.cpp

```diff
@@ -504,6 +504,26 @@ bool generate_ansi_c_start_function(
 
   record_function_outputs(symbol, init_code, symbol_table);
 
+  // now call destructor functions (a GCC extension)
+
+  for(const auto &symbol_table_entry : symbol_table.symbols)
+  {
+    const symbolt &symbol = symbol_table_entry.second;
+
+    if(symbol.type.id() != ID_code)
+      continue;
+
+    const code_typet &code_type = to_code_type(symbol.type);
+    if(
+      code_type.return_type().id() == ID_destructor &&
+      code_type.parameters().empty())
+    {
+      code_function_callt function_call(symbol.symbol_expr());
```

Comment:
Tiny nit pick ;-) but I'd probably name this something like 'destructor_call' just to make it super explicit what/why we're creating and inserting this call - but that's probably just because my attention span is tiny :-)